app/services/post_alerter.rb

```diff
@@ -583,8 +583,11 @@ def extract_mentions(post)
   # TODO: Move to post-analyzer?
   # Returns a list of users who were quoted in the post
   def extract_quoted_users(post)
-    usernames = post.raw.scan(/\[quote=\"([^,]+),.+\"\]/).uniq.map { |q| q.first.strip.downcase }
-
+    usernames = if SiteSetting.display_name_on_posts && !SiteSetting.prioritize_username_in_ux
+      post.raw.scan(/username:([[:alnum:]]*)"/)
```

Comment:
Shouldn't this look for `username: …` **inside** a `[quote=…` block? Otherwise it can catch unrelated "username: " text

Reply:
I updated the regex to only search within the quote block

https://github.com/discourse/discourse/pull/17030/files#diff-0f98de8fdfe3e956c826cb7bb7923bc461951bbbe2fc7fb78117f78fff263103R587

and added it to the spec

https://github.com/discourse/discourse/pull/17030/files#diff-80e924e317914a40c584ededc9ce8023cc9d715d4536b35ea8c7c0fde4d15a4bR456